pandas/core/arrays/base.py

```diff
@@ -45,11 +51,18 @@ def try_cast_to_ea(cls_or_instance, obj, dtype=None):
     -------
     ExtensionArray or obj
     """
-    try:
-        result = cls_or_instance._from_sequence(obj, dtype=dtype)
-    except Exception:
-        # We can't predict what downstream EA constructors may raise
+    if (
+        is_integer_dtype(cls_or_instance)
```

Comment:
what exactly are you trying to select in here?

Reply:
If the datatype of `cls_or_instance(self._values passed from combine function)` is an integer type and the `obj(new_values passed by the combine function)` is of type bool then it should not be converted to EA because it will then create EA of ints instead of bools. And if the `cls_or_instance` is of type sparse then also it shouldn't convert it to EA because it should return a sparse array instead of normal EA.

Reply:
Do you have any hint why it gives the ImportError?

Reply:
>If the datatype of cls_or_instance(self._values passed from combine function) is an integer type and the obj(new_values passed by the combine function) is of type bool then it should not be converted to EA because it will then create EA of ints instead of bools. And if the cls_or_instance is of type sparse then also it shouldn't convert it to EA because it should return a sparse array instead of normal EA.

I am sorry! That explanation is a  bit wrong. 

it is like,
If the datatype of self._values is integer type and the new_values is of type bool then it should not be converted to EA. But the sparse arrays having datatype of self._values as integer and the new_values as bool should be converted to EA because the returned EA should be sparse also. So the condition is like(See in my last commit),
```
if not (
                is_integer_dtype(self._values)
                and is_bool(new_values[0])
                and not isinstance(self._values, SparseArray)
            ):
                new_values = try_cast_to_ea(self._values, new_values)
```.

Reply:
@rushabh-v I don't fully understand why this is needed / how this is related to or caused by the other changes in this PR. 

Can you show the test or example that is failing if you don't do the above change?

Reply:
https://github.com/pandas-dev/pandas/blob/a72eef5a86979232337149961d2122f0b555c876/pandas/tests/extension/base/ops.py#L29

This test fails if we don't put those conditions and it says:
```
AssertionError: Attributes of Series are different           
           Attribute "dtype" are different
           [left]:  bool
           [right]: Int64
```